Tools/pyco/pyco.py

```diff
@@ -239,6 +253,7 @@ def rewritten_bytecode(code: types.CodeType, builder: Builder) -> bytes:
                 raise RuntimeError(
                     f"More than 256 constants in original {code.co_name} at line {code.co_firstlineno}"
                 )
+                oparg = oparg | (instrs[i - 1] << 8)
```

Comment:
So this is unreachable... Which is just as well, since the code below would still duplicate the EXTENDED_ARG op.

Anyway, it looks like for co_names you stumbled upon the right approach, which is not to rewrite the bytecode at all, instead ensuring that the new index always matches the original index. (Do you check for that below?)

Eventually, when you get to this for co_consts, we'll need to decide whether to still rewrite the code to optimize immediate constants. If we do, we'll have to make sure that things work right if an instruction with an immediate constant has an EXTENDED_ARG prefix. (We'd have to replace that with a NOP.)

(Also, in theory there can be more than one EXTENDED_ARG prefix, though that would require a truly gargantuan function -- we can ignore those in the prototype.)

Reply:
> So this is unreachable... 

Yes, our tests currently avoid it